spec/index.bs

```diff
@@ -1067,6 +1067,26 @@ dictionary GPULimits {
 };
 </script>
 
+#### <dfn interface>GPUAdapterLimits</dfn> ### {#gpu-adapterlimits}
+
+{{GPUAdapterLimits}} exposes the [=limits=] supported by an adapter.
+See {{GPUAdapter/limits|GPUAdapter.limits}}.
+
+<script type=idl>
+interface GPUAdapterLimits {
```

Comment:
do we have to redefine the whole dictionary here? can we re-use `GPULimits` for the adapter, instead?

Reply:
1. It seems like it should be an interface, as a dictionary-returning attribute creates a new dictionary object and returns it every time you call it (I think).
2. I'm pretty sure GPULimits can soon be removed after doing the TODO about `record<>`.

Reply:
1) Not with [SameObject] surely!

Reply:
I thought about it, but we want it for the attributes behavior, but also:

> The [SameObject] extended attribute must not be used on anything other than a read only attribute whose type is an interface type or object.

Reply:
(idk why it works on `object` but not dictionaries, but that's what it says)

Reply:
> 2\. I'm pretty sure GPULimits can soon be removed after doing the TODO about `record<>`.

I looked into doing this, and it's unfortunately not so simple. GPULimits provides type checking and defaulting that we would otherwise have to write out explicitly.

Unfortunately we still need to do something to ensure UAs don't ignore keys they don't understand. So instead I'm thinking we can have steps like this:

- Convert the record to GPULimits
- Check for fields other than the known fields
- If either of those fails, fail device creation

Given this, it could make sense to just have the attribute type be `GPULimits`. But WebIDL:

> The type of the attribute, after resolving typedefs, must not be a nullable or non-nullable version of any of the following types: ..., a dictionary type, ...

So the attribute type would have to be `object`.

Is it better to have the attribute type be `object` or to give it an explicit `interface` type?

Reply:
Will resolve in editorial followup.